### PR TITLE
Fix the issue that ar target is missing in the generated Makefile

### PR DESCRIPTION
**Pull request by tomflinda**

### CODE REVIEW

**Comment by tomflinda** [comment]
The two failres are known issues, not related to this PR.

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/GenMakefile.cpp [hunk 1]
```diff
@@ -77,26 +77,39 @@ static void getCompileInfo(
...
         } else if (llvm::StringRef(Obj).endswith(".o")) {
           llvm::SmallString<512> FilePathAbs(Obj);
           llvm::sys::path::native(FilePathAbs);
           llvm::sys::fs::make_absolute(FilePathAbs);
           llvm::sys::path::remove_dots(FilePathAbs, true);
-          ObjsInLinkerCmd[std::string(FilePathAbs.str())] = true;
-          ObjsInLinker[std::string(FilePathAbs.str())] = true;
+          ObjsInLKOrARCmd.push_back(std::string(FilePathAbs.str()));
           ObjName = std::string(FilePathAbs.str());
+        } else if (Obj == "ar") {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/GenMakefile.cpp:107]: could you add comment for the new added logic.

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
Also could you try to add a lit test for the issue.

File: clang/lib/DPCT/GenMakefile.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/GenMakefile.cpp:107]: okay

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**